cobol-parser/src/main/scala/za/co/absa/cobrix/cobol/parser/decoders/StringDecoders.scala

```diff
@@ -22,29 +22,45 @@ import scala.util.control.NonFatal
 
 object StringDecoders {
 
+  // Simple constants are used instead of enumeration for better performance
+  val TrimNone  = 1
+  val TrimLeft  = 2
+  val TrimRight = 3
+  val TrimBoth  = 4
+
   /**
     * A decoder for any EBCDIC string fields (alphabetical or any char)
     *
-    * @param bytes A byte array that represents the binary data
+    * @param bytes        A byte array that represents the binary data
+    * @param trimmingType Specifies if and how the soutput string should be trimmed
     * @return A string representation of the binary data
     */
-  def decodeEbcdicString(bytes: Array[Byte]): String = {
+  def decodeEbcdicString(bytes: Array[Byte], trimmingType: Int): String = {
     var i = 0
     val buf = new StringBuffer(bytes.length)
     while (i < bytes.length) {
       buf.append(BinaryUtils.ebcdic2ascii((bytes(i) + 256) % 256))
       i = i + 1
```

Comment:
maybe something more functional like `bytes.zipWithIndex.foreach` ...?

Reply:
Yes, it is a good suggestion. I look at imperative code similar as well. However, decoders are on critical performance path and imperative code provides much better performance than functional style. So I've deliberately rewritten this part of the program to use while loops and vars.